components/MainCategories.tsx

```diff
@@ -0,0 +1,130 @@
+import styled, { css } from 'styled-components';
+import CategoryIcon from './CategoryIcon';
+import categories from '@/lib/categories.json';
+import { useState } from 'react';
+import Arrow from '@/public/icons/categories/Arrow.svg';
+import {
+  MainCategoriesProps,
+  MainCategoriesStyledComponents,
+} from '@/types/MainCategories.types';
+
+export default function MainCategories({
+  onChangeFilter,
+  activeFilter,
+}: MainCategoriesProps) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <StyledWrapper>
+      <StyledFilterGroup $isOpen={isOpen}>
+        <StyledToggleGroup onClick={() => setIsOpen(!isOpen)}>
+          <StyledH2>{isOpen ? 'Close Categories' : 'Open Categories'}</StyledH2>
+          <StyledArrowDown $isOpen={isOpen} />
+          <StyledArrowUp $isOpen={isOpen} />
+        </StyledToggleGroup>
+        <StyledUl $isOpen={isOpen}>
+          {categories.map((category) => (
+            <li key={category.id}>
+              <CategoryIcon
+                category={category}
+                onChangeFilter={onChangeFilter}
+                activeFilter={activeFilter}
+              />
+            </li>
+          ))}
+        </StyledUl>
+      </StyledFilterGroup>
+    </StyledWrapper>
+  );
+}
+
+const StyledWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+`;
+
+const StyledFilterGroup = styled.div<MainCategoriesStyledComponents>`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  background-color: var(--color-primary-1);
+  border-radius: var(--radius-m);
+  width: 325px;
+  transition: all 0.6s ease-in-out;
+  overflow: hidden;
+
+  ${({ $isOpen }) =>
+    $isOpen &&
+    css`
+      max-height: 350px;
+    `}
+
+  ${({ $isOpen }) =>
+    !$isOpen &&
+    css`
+      max-height: 35px;
+    `}
+`;
+
+const StyledUl = styled.ul<MainCategoriesStyledComponents>`
+  display: grid;
+  gap: var(--spacing-2);
+  grid-template-columns: repeat(5, auto);
+  justify-content: space-between;
+  padding-inline: var(--spacing-3);
+  transition: all 0.6s ease-in-out;
+  overflow: hidden;
+  ${({ $isOpen }) =>
+    $isOpen &&
+    css`
+      max-height: 300px;
+      opacity: 1;
+      padding-block: var(--spacing-3);
+    `}
+  ${({ $isOpen }) =>
+    !$isOpen &&
+    css`
+      max-height: 50px;
+      opacity: 0;
+      padding-block: 0;
+    `}
+`;
```

Comment:
same is true for these other elements.

Reply:
By removing this, I would get rid of the "fading menu"